packages/api/src/controllers/eventController.ts

```diff
@@ -0,0 +1,99 @@
+import {Request, Response} from 'express';
+import Event from '../models/event';
+import { validateJson } from '../validators/eventValidator';
+
+export const getAllEvents = async (req: Request, res: Response) =>
+{
+  const events = await Event.find();
+
+  return res.status(200).json({
+    "events": events,
+    "message": "Events have been retrieved succesfully."
+  })
+}
+
+export const createEvent = async(req: Request, res: Response) =>
+{
+  if (validateJson(req.body)) {
+    const eventData = req.body;
+    const event = await Event.create(eventData);
+
+    return res.status(200).json({
+      "event": event,
+    })
+  }
+  else {
+    return res.status(400).json({
+      "event": req.body,
+      "message": "Invalid JSON was passed."
+    })
+  }
+}
+
+
+export const getEventById = async(req: Request, res: Response) =>
+{
+  const eventId = req.params.id;
+  const event = await Event.findOne({_id: eventId});
+
+  if (event == null) {
+    return res.status(500).json({
+      "message": "No event was found."
+    })
+  }
+
+  return res.status(200).json({
+    "event": event,
+  })
```

Comment:
Makes more sense to use Promise style syntax here to make it clenear.
```js
Event.findOne({_id: eventId}).exec()
  .then(event => {
    return res.status(200).json({"event": event});
  })
  .catch(err => {
    return res.status(404).json({"message": err});
  }
);
```

Reply:
Fixed